Sources/Fuzzilli/Fuzzer.swift

```diff
@@ -486,21 +498,46 @@ public class Fuzzer {
 
     /// Process a program that has interesting aspects.
     func processInteresting(_ program: Program, havingAspects aspects: ProgramAspects, origin: ProgramOrigin) {
+        var aspectsToTrack = aspects
+
         func processCommon(_ program: Program) {
             let (newTypeCollectionRun, _) = updateTypeInformation(for: program)
 
             dispatchEvent(events.InterestingProgramFound, data: (program, origin, newTypeCollectionRun))
 
             // All interesting programs are added to the corpus for future mutations and splicing
-            corpus.add(program)
+            corpus.add(program, aspectsToTrack)
+        }
+
+        // If only adding deterministic samples, execute each sample additional times to verify determinism
+        // Each sample will be executed at least minDeterminismExecs, and no more than maxDeterminismExecs times
+        // If two consecutive executions return the same edges after at least minDeterminismExecs times, the sample
+        // is considered deterministic 
+        if deterministicCorpus {
+
+            var foundSameAspects: Bool
+            var newAspects : ProgramAspects = aspects
+            var rounds = 0
+
+            repeat {
+                evaluator.resetAspects(newAspects)
+                let execution = execute(program)
+                guard execution.outcome == .succeeded else { return }
+
+                guard let res = evaluator.evaluateAndIntersect(execution, with: newAspects) else { return }
```

Comment:
I think it would be a bit more intuitive here to have this method only return the intersected aspects and then performing the `newAspects == previousAspects` manually here? E.g.
```
guard let intersectedAspects = evaluator.evaluateAndIntersect(execution, with: aspects) else { return }
didConverge = intersectedAspects == aspects
aspects = intersectedAspects
```
WDYT?

Reply:
I think that because `Set` isn't ordered, this line [here](https://github.com/googleprojectzero/fuzzilli/blob/6e11b776b0582c5664ab31bef1cc466fef75cda8/Sources/Fuzzilli/Evaluation/ProgramCoverageEvaluator.swift#L228) results in `secondCovEdgeSet` being a different order than the original, and the comparison in `CovEdgeSet` assumes sorted sets.

Since the finding new coverage is supposed to be a rare event, and the list of new edges should be relatively short, I'm thinking that sorting and then doing the comparison as described makes sense. 

My next commit will include that functionality as it should be an easy change, although I'd appreciate your thoughts on this.

Reply:
Whats the preferred way of handling the aspect comparison? My understanding is that it isn't possible a static comparison method in `ProgramAspects` that is overridden by one in `CovEdgeSet`, without having to downcase in Fuzzer.swift.

Am I missing something?